library.go

```diff
@@ -1043,23 +1043,33 @@ func zip(thread *Thread, _ *Builtin, args Tuple, kwargs []Tuple) (Value, error)
 		}
```

Comment:
Don't forget to call Done on all the iterators on all control-flow paths---like your original CL did. :)

For some reason GitHub won't let me comment on L1040 because it's too far away from the diff chunk.

Reply:
I kept the fix for Done as a separate PR: https://github.com/google/skylark/pull/5 (implemented using defer as you suggested). If you merge it first I will rebase this one and address all the issues you mentioned below. Thanks!

Reply:
Ah, I see.  I just committed that change for you.

Reply:
All done, please have a look